src/tic_tac_toe/session.clj

```diff
@@ -0,0 +1,47 @@
+(ns tic-tac-toe.session
+  (:require [tic-tac-toe.display :refer :all]
+            [tic-tac-toe.board :refer :all]
+            [tic-tac-toe.player :refer :all]
+            [tic-tac-toe.rules :refer :all]
+            [tic-tac-toe.messages :refer :all]))
+
+(defn validate-integer [entry]
+  (and (number? entry) (>= entry 1) (<= entry 9)))
+
+(defn empty-space? [board move]
+  (number? (nth board (- move 1))))
+
+(defn valid-move? [board move]
+  (and (validate-integer move) (empty-space? board move)))
```

Comment:
It seems like this session namespace has more than one responsibility. For example, I'm not sure if the valid-move? function belongs here.

Reply:
mmmm do you suggest I make separate file that validates moves? Or do you think it makes sense to put it in rules?